### PR TITLE
Introduce `Limit` type

### PR DESCRIPTION
**Pull request by christophstrobl**
`Limit` can be used as `Parameter` within repository query methods to define the maximum number of results to be returned. 

```java
List<Person> findByFirstnameLike(String firstname, Limit limit);

List<Person> findByFirstnameLike(String firstname, Sort orderBy, Limit limit);

Window<User> findByFirstnameLikeOrderByLastname(ScrollPosition pageable, Limit limit);
```

The default `ParameterAccessor` implementation will pick up `Limit` and `Sort` to assemble a `Pageable` out of the provided values if present to provide out of the box support for modules using `ParameterAcessor#getPageable()` to apply order and limit.

`Limit#max` is aligned to `Page.size`, both using `int`. We might want to consider `long` for `Limit` and error if the actual value exceeds the `Page` one when assembling the `Pageable`.

### CODE REVIEW

File: src/main/asciidoc/repositories-paging-sorting.adoc [hunk 1]
```diff
@@ -57,6 +61,28 @@ Rather, it restricts the query to look up only the given range of entities.
...
+| `findBy...(Pageable page, Sort sort)`
+| `Pageable` already defines `Sort`
+
+| `Pageable` & `Limit`
+| `findBy...(Pageable page, Limit limit)`
+| `Pageable` already defines a limit.
+
+|===
+
+The `Top` keyword used to limit results can be used to along with `Pageable` whereas `Top` defines the total maximum of results, whereas the Pageable parameter may reduce this this number.
```
Comment (odrotbohm) [inline src/main/asciidoc/repositories-paging-sorting.adoc:83]: Double "this this"?

File: src/main/java/org/springframework/data/repository/query/QueryMethod.java [hunk 2]
```diff
@@ -290,6 +264,41 @@ public String toString() {
 		return method.toString();
 	}
 
+	public void validate() {
```
Comment (odrotbohm) [inline src/main/java/org/springframework/data/repository/query/QueryMethod.java:267]: Needs to be public?

**Review by odrotbohm — COMMENTED** [review]
A couple of minor findings.

**Review by odrotbohm — COMMENTED** [review]
A couple of minor findings.

**Review by mp911de — CHANGES_REQUESTED** [review]
Left some comments, seconding what Ollie said.

**Comment by mp911de** [comment]
That's merged and polished now.

**Pull request closed without merging**